src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Models/DemoCustomerOrderEntity.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections.ObjectModel;
+using System.Linq;
+using VirtoCommerce.DemoSolutionFeaturesModule.Core.Models;
+using VirtoCommerce.OrdersModule.Core.Model;
+using VirtoCommerce.OrdersModule.Data.Model;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.DemoSolutionFeaturesModule.Data.Models
+{
+    public class DemoCustomerOrderEntity: CustomerOrderEntity
+    {
+        public virtual ObservableCollection<DemoOrderConfiguredGroupEntity> ConfiguredGroups { get; set; } = new NullCollection<DemoOrderConfiguredGroupEntity>();
+
+        public override OrderOperation ToModel(OrderOperation operation)
+        {
+            var orderExtended = (DemoCustomerOrder)base.ToModel(operation);
+
+            orderExtended.ConfiguredGroups =
+                ConfiguredGroups
+                    .Select(x => x.ToModel(AbstractTypeFactory<DemoOrderConfiguredGroup>.TryCreateInstance()))
+                    .ToList();
+
+            return orderExtended;
+        }
+
+        public override OperationEntity FromModel(OrderOperation operation, PrimaryKeyResolvingMap pkMap)
+        {
+            base.FromModel(operation, pkMap);
+
+            var orderExtended = (DemoCustomerOrder) operation;
```

Comment:
I think it would be better to replace cast by if statement.
like this:
if (operation is DemoCustomerOrder orderExtended)
{
...
}

What do you think?

Reply:
We guarantee we always will pass model or it's successor here so it's redundant to add additional nesting level with if